src/Socket.js

```diff
@@ -0,0 +1,109 @@
+import { Socket as PhoenixSocket } from 'phoenix';
+
+export const states = {
+  CONNECTING: 'CONNECTING',
+  CONNECTED: 'CONNECTED',
+  JOINING: 'JOINING',
```

Comment:
As mentioned below, this could be actually "JOINED" or something like "WAITING" instead.

Reply:
Good idea!